packages/gatsby/src/joi-schemas/joi.js

```diff
@@ -34,10 +34,14 @@ export const gatsbyConfigSchema = Joi.object()
     }).unknown(),
     mapping: Joi.object(),
     plugins: Joi.array(),
-    proxy: Joi.object().keys({
-      prefix: Joi.string().required(),
-      url: Joi.string().required(),
-    }),
+    proxy: Joi.array()
+      .items(
+        Joi.object().keys({
+          prefix: Joi.string().required(),
+          url: Joi.string().required(),
+        })
+      )
+      .single(),
```

Comment:
It might be my dev preference. Having a more specific type either `array` or `object` gives more predictability to the code execution. I would rather release it as a breaking change.

Having only:
```javascript
proxy: Joi.array()
      .items(
        Joi.object().keys({
          prefix: Joi.string().required(),
          url: Joi.string().required(),
        })
      ),
```